packages/vite/src/node/plugins/resolve.ts

```diff
@@ -725,8 +725,7 @@ export function resolvePackageEntry(
         }
       }
     }
-
-    entryPoint = entryPoint || data.main || 'index.js'
+    entryPoint = entryPoint || data.main || 'index'
```

Comment:
This would mean that by default, [extensions like `mjs`, `tsx`, etc](https://github.com/vitejs/vite/blob/133fcea5223263b0ae08ac9a0422b55183ebd266/packages/vite/src/node/constants.ts#L9), will be checked too. I'm not sure if this is good [spec-wise](https://nodejs.org/api/modules.html#all-together), since it only checks `js`, `json`, and `node` (which isn't covered in this PR).

Perhaps to fix this, we can set `options.extensions` as `['js','json','node']`, which is [passed](https://github.com/vitejs/vite/blob/133fcea5223263b0ae08ac9a0422b55183ebd266/packages/vite/src/node/plugins/resolve.ts#L342) into `tryFsResolve()`?

Reply:
Actually, I think this could possibly bring a breaking change? If `data.main` was specified, and was relied to load `${data.main}/index.mjs`, it wouldn't work anymore, but could be an edge case.

Reply:
Should js/json/node be the default inspection item?

Reply:
I think js/json/node should be the only ones iff we're falling back to `index`. For other cases, it would be safer to leave it as is. But maybe the VIte team do want to support more than those 3, which I'm fine as well.